pkg/resources/custom_diffs.go

```diff
@@ -195,3 +195,32 @@ func RecreateWhenUserTypeChangedExternally(userType sdk.UserType) schema.Customi
 		return nil
 	}
 }
+
+func RecreateWhenSecretTypeChangedExternally(secretType string, oauthType *sdk.OauthSecretType) schema.CustomizeDiffFunc {
```

Comment:
The setup of this func is a bit convoluted and delegated to the user's invocation. We know which resources result in which type and what additional logic is needed. Let's move this logic inside the function. To do that, let's have all 4 values of secret type as input and adjust the logic properly.

Reply:
ok i did something similar to what is done for the user_type, that way the invocation is done only by providing one argument, one of the four types of creating the secret